components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/dao/ParMgtDAOImpl.java

```
@@ -0,0 +1,134 @@
+                    String clientId = resultSet.getString(ParConstants.COL_LBL_CLIENT_ID);
+
+                    return new ParRequestDO(getParams(jsonParams), scheduledExpiry, clientId);
+
+                } else {
+                    throw new ParCoreException("uuid does not exist in the database");
+                }
+            } catch (SQLException e) {
+                throw new ParCoreException("Error occurred while retrieving PAR request from the database.");
+            }
+        } catch (SQLException e) {
+            throw new ParCoreException("Error occurred while retrieving PAR request from the database.");
+        }
+    }
+
+    @Override
+    public void removeParRequest(String reqUUID) throws ParCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+             PreparedStatement prepStmt = connection.prepareStatement(SQLQueries
+                     .ParSQLQueries.REMOVE_PAR_REQUEST)) {
+            prepStmt.setString(1, reqUUID);
+            prepStmt.execute();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            throw new ParCoreException("Error occurred while clearing PAR request from Database");
+        }
+    }
+
+    private String getJsonParams(Map<String, String> params) throws ParCoreException {
+
+        try {
+            return objectMapper.writeValueAsString(params);
+        } catch (JsonProcessingException e) {
+            throw new ParCoreException("Error occurred while serializing parameter map to JSON");
+        }
+    }
+
+    private Map<String, String> getParams(String jsonParams) throws ParCoreException {
+
+        try {
+            return objectMapper.readValue(jsonParams, new TypeReference<Map<String, String>>() { });
+        } catch (JsonProcessingException e) {
+            throw new ParCoreException("Error occurred while serializing JSON string map to Map");
```

the actual exception is dropped. Through the actual exception as well.